### ISSUE TITLE
Re-introduce miniLM embedding index

### ISSUE THREAD

**Issue by mjmor** [greenearth-social/ingex#473]
- maybe more RAM
- also give Juan metrics to watch and times to run experiments

**Comment by inthree3** [comment]
Just out of curiosity, was the previous MiniLM embedding removed because of memory usage issues?

**Comment by inthree3** [comment]
@juan-lievano 

I wanted to check if there is already an implementation plan for the semantic feed control feature in GreenEarth. Also, I'm slightly unclear on whether this is meant for candidate generation (CG), ranking, or both.

Here is my current understanding of the intended flow:
1. A trained model generates a query vector from a natural language prompt (384-dim, matching MiniLM).
2. The API calls `knn_search_posts(..., search_field=MINILM_L12_EMBEDDING_FIELD, query_vector=...)`.
3. Elasticsearch traverses the HNSW graph on `all_MiniLM_L12_v2` to find the nearest neighbors.
4. The most similar posts are returned.

I may have missed some context—could you clarify if this aligns with what you had in mind?

**Comment by mjmor** [comment]
> Just out of curiosity, was the previous MiniLM embedding removed because of memory usage issues?

Yes it was @inthree3 , but we've increased since then a couple times already. So let's try re-enabling and see what happens 🤷 and ith the dashboard you added in your PR plus the perf profiling dashboard (which tracks memory usage and page cache churn) we should be able to confirm if RAM is scarce after this change.

**Comment by juan-lievano** [comment]
@inthree3 Yes your description is what I had in mind. I might reword step 1 to be something like "Given a prompt, an LLM scores how relevant a small sample of posts is with respect to that prompt, and a vector is fit to approximate those LLM scores". That vector is then the query vector used in steps 2, 3 and 4, as you described.

Let me know if this makes sense!

**Comment by juan-lievano** [comment]
@inthree3 About your first question: currently this a candidate generator but it actually also produces a ranking so we could test if works well as a heavy ranker too. 

(Although the point of using a query vector is to get sublinear time complexity so that we can use it as a candidate generator, perhaps if we want an LLM heavy ranker we could use an actual LLM and not an approximation of one!)

**Comment by inthree3** [comment]
@juan-lievano Makes sense. It's $$v_q =\arg\min_{v} \left( \sum_{i \in \text{sample}} |v \cdot e_i - s_i| + \lambda \|v\|_2^2 \right)$$ (I asked Claude to make this equation)

**Comment by juan-lievano** [comment]
@inthree3 Yes! 

Some tiny caveats: 1. we might use absolute value instead of taking the square (i.e. L1 norm instead of L2 norm) and 2. there will be some regularization (i.e. a term in that formula that "punishes" vectors v_q that have coefficients that are too big) to prevent overfitting.

But those are technicalities, the mental picture your equation gives is totally appropriate.

**Comment by inthree3** [comment]
@juan-lievano I like the idea of using the LLM directly in the heavy ranker phase. That makes sense, and we'll just need to do some more experiments with different-sized models, batch sizes, and latency.
